File: src/main/java/net/sileader/cyborg/basys/Basys.java
package net.sileader.cyborg.basys;

import com.sun.jna.Library;
import com.sun.jna.Native;

public interface Basys extends Library {
    Basys INSTANCE= Native.loadLibrary("basys", Basys.class);


    void setUid(int uid);
    int getUid();

    int changeRootDirectory(String path);

    int forkProcess();
    int waitProcess();
    int waitPidProcess(int pid);
}

File: src/main/kotlin/net/sileader/cyborg/core/Basys.kt
package net.sileader.cyborg.core

import net.sileader.cyborg.basys.Basys

object Basys {
    var uid: Int
        get() = Basys.INSTANCE.uid
        set(value) {
            Basys.INSTANCE.uid=value
        }

    fun changeRootDirectory(path: String): Int = Basys.INSTANCE.changeRootDirectory(path)

    fun forkProcess(): Int = Basys.INSTANCE.forkProcess()
    fun waitProcess(): Int = Basys.INSTANCE.waitProcess()
    fun waitPidProcess(pid: Int): Int = Basys.INSTANCE.waitPidProcess(pid)
}